src/dns_provider/mod.rs

```diff
@@ -1,12 +1,38 @@
 use std::net::Ipv4Addr;
 
+use crate::dns_provider::cloudflare::CloudflareDnsConfig;
 use crate::dns_provider::errors::DnsProviderError;
+use crate::dns_provider::qoverydns::QoveryDnsConfig;
 use serde::{Deserialize, Serialize};
+use tera::Context as TeraContext;
 
 use crate::io_models::{Context, Domain};
 
 pub mod cloudflare;
 pub mod errors;
+pub mod qoverydns;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum Kind {
```

Comment:
OPT: split `io` from `domain` models avoiding future issues.
CF https://github.com/Qovery/engine/blob/dev/src/cloud_provider/io.rs#L6 and 
https://github.com/Qovery/engine/blob/dev/src/cloud_provider/mod.rs#L51

Domain struct shouldn't derive Serialize and Deserialize.